source/grid/tria_description.cc

```diff
@@ -955,30 +955,39 @@ namespace TriangulationDescription
 
       std::vector<LinearAlgebra::distributed::Vector<double>> partitions_mg;
 
-      if (construct_multigrid) // perform first child policy
+      // If desired, also create a multigrid hierarchy. For this, we have to
+      // build a hierarchy of partitions (one for each level of the
+      // triangulation) in which each cell is assigned to the same process
+      // as its first child (if not active) or to the same process that already
+      // owns the cell (for an active level-cell).
+      if (construct_multigrid)
         {
           const auto tria_parallel =
             dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
               &tria);
+          Assert(tria_parallel, ExcNotImplemented());
 
-          Assert(tria_parallel, ExcInternalError());
-
-          partition.update_ghost_values();
-
+          // Give the level partitioners the right size:
           partitions_mg.resize(tria.n_global_levels());
-
           for (unsigned int l = 0; l < tria.n_global_levels(); ++l)
             partitions_mg[l].reinit(
               tria_parallel->global_level_cell_index_partitioner(l).lock());
 
+          // Make sure we know about all of the owners of the active cells,
+          // whether locally owned or not. Then we traverse the triangulation
+          // from the finest level to the coarsest level:
+          partition.update_ghost_values();
           for (int level = tria.n_global_levels() - 1; level >= 0; --level)
             {
+              // On each level, traverse the cell. If the cell is not locally
+              // owned, we don't care about it. If it is active, we copy the
+              // owner process from the cell's non-level owner. Otherwise,
+              // use the owner of the first cell
               for (const auto &cell : tria.cell_iterators_on_level(level))
                 {
                   if (cell->is_locally_owned_on_level() == false)
                     continue;
```

Comment:
I think you generally prefer to avoid `continue` statements and instead wrap the rest into the `if` (which would be ok here), so I was surprised you left this in. :wink: Again, nothing wrong and nothing to be changed.

Reply:
I'm lenient towards my co-authors in places where everything fits into <10 lines of code, but I have done as suggested :-)